api/src/api/members.js

```diff
@@ -0,0 +1,114 @@
+const express = require('express');
+const router = express.Router();
+const Member = require('./../models/member');
+const { levelEnum } = Member;
+const errorWrap = require('../middleware/errorWrap');
+const { requireRegistered, requireDirector } = require('../middleware/auth');
+const { filterSensitiveInfo } = require('../utils/user-utils');
+
+const validateMemberQuery = (req, res, next) => {
+  // Middleware that verifies that fields to be inserted/updated actually
+  // exist in the Member schema
+  for (const field in req.body) {
+    if (!Member.schema.paths[field]) {
+      return res.status(400).json({
+        success: false,
+        message: `no such field ${field} in member`,
+      });
+    }
+  }
+  next();
+};
+
+router.get(
+  '/current',
+  errorWrap(async (req, res) => {
+    res.json({
+      success: true,
+      result: req.user ? filterSensitiveInfo(req.user) : null,
```

Comment:
Is this for getting the information of the current user? If so, don't we need to fetch the member data from the DB by oauthID?

Reply:
Nevermind, I figured it out